### PR TITLE
Add odometry publisher system

### LINKED ISSUE #465: Odometry for omnidirectional robot
Hi,  I am using ignition gazebo: citadel version to simulate an omnidirectional robot (3 wheels). 

Using gazebo I can control the robot with the plugin libgazebo_ros_planar_move which allows me to define the command and odometry topics, however this plugin is not available on ignition gazebo. 
Is there any similar plugin I could use?. Currently I can make the robot to move using the ignition-gazebo-velocity-control-system plugin, but I am not able to get odometry out of it. The only plugin I have seen with odometry output is diff_drive_system which is not my case.

I appreciate any help.

### PR DESCRIPTION
**Pull request by mrushyendra**
Adds a new system that publishes odometry messages for an entity, in a manner similar to what the `ros_planar_move` plugin for Gazebo does. It assumes the robot is on a flat plane and publishes 2D position and velocity data.

There may be other requirements for such a system that I might have missed out on adding, so I can make any changes or additions if necessary.

Closes #465

### CODE REVIEW

File: src/systems/odometry_publisher/OdometryPublisher.cc [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  if (odomFreq > 0)
+  {
+    std::chrono::duration<double> period{1 / odomFreq};
+    this->dataPtr->odomPubPeriod =
+      std::chrono::duration_cast<std::chrono::steady_clock::duration>(period);
+  }
+
+  // Setup odometry
+  std::string odomTopic{"/model/" + this->dataPtr->model.Name(_ecm) +
+    "/odometry"};
```
Comment (ahcorde) [inline src/systems/odometry_publisher/OdometryPublisher.cc:149]:
```suggestion
  std::string odomTopic{transport::TopicUtils::AsValidTopic("/model/" + this->dataPtr->model.Name(_ecm) +
    "/odometry")};
  if (odomTopic.empty())
  {
    ignerr << "Failed to generate odom topic ["
           << this->dataPtr->model.Name(_ecm) << "]" << std::endl;
    return;
  }
```

File: src/systems/odometry_publisher/OdometryPublisher.hh [hunk 2]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
```
Comment (ahcorde) [inline src/systems/odometry_publisher/OdometryPublisher.hh:2]:
```suggestion
 * Copyright (C) 2021 Open Source Robotics Foundation
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/systems/odometry_publisher/OdometryPublisher.cc [hunk 3]
```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
```
[the remark this answers is not in the document]
Comment (mrushyendra) [reply src/systems/odometry_publisher/OdometryPublisher.cc]: Ah yes, of course...

**Review by ahcorde — CHANGES_REQUESTED** [review]
Can you add some tests ? have a look [here](https://github.com/ignitionrobotics/ign-gazebo/tree/ign-gazebo4/test/integration)

**Review by ahcorde — CHANGES_REQUESTED** [review]
LGTM, but do you mind to solve the conflics ?

**Comment by ahcorde** [comment]
friendly ping @mrushyendra

**Comment by ahcorde** [comment]
@chapulina do you mind to have a quick look here?

**Review by ahcorde — APPROVED** [review]

**Pull request merged**